Author: Anjandev Momi <anjan@momi.ca>
URL: https://github.com/slgobinath/SafeEyes/pull/485
Summary: disable bindtextdomain, alpine's python does not support it. Error without this patch: AttributeError: module 'locale' has no attribute 'bindtextdomain'
----
commit 96b3477ed9bdd12bcf9044268f2be9c35bbbbda6
Author: Anjandev Momi <anjan@momi.ca>
Date:   Sun Oct 2 19:27:29 2022 -0400

    add support for python without gettext.bindtextdomain

diff --git a/safeeyes/__main__.py b/safeeyes/__main__.py
index 212631f..524ad96 100755
--- a/safeeyes/__main__.py
+++ b/safeeyes/__main__.py
@@ -90,9 +90,12 @@ def main():
     """
     system_locale = gettext.translation('safeeyes', localedir=utility.LOCALE_PATH, languages=[utility.system_locale(), 'en_US'], fallback=True)
     system_locale.install()
-    # locale.bindtextdomain is required for Glade files
-    # gettext.bindtextdomain(gettext.textdomain(), Utility.LOCALE_PATH)
-    locale.bindtextdomain('safeeyes', utility.LOCALE_PATH)
+    try:
+        # locale.bindtextdomain is required for Glade files
+        locale.bindtextdomain('safeeyes', utility.LOCALE_PATH)
+    except AttributeError:
+        logging.warning('installed python\'s gettext module does not support locale.bindtextdomain. locale.bindtextdomain is required for Glade files')
+
 
     parser = argparse.ArgumentParser(prog='safeeyes', description=_('description'))
     group = parser.add_mutually_exclusive_group()
